ggml/src/ggml-metal/ggml-metal.m

```diff
@@ -5892,25 +5878,117 @@ static void ggml_backend_metal_buffer_free_buffer(ggml_backend_buffer_t buffer)
 }
 
 static void ggml_backend_metal_buffer_memset_tensor(ggml_backend_buffer_t buffer, struct ggml_tensor * tensor, uint8_t value, size_t offset, size_t size) {
+#if 1
+    struct ggml_backend_metal_buffer_context * ctx = (struct ggml_backend_metal_buffer_context *)buffer->context;
+
+    @autoreleasepool {
+        id<MTLCommandQueue>       queue   = ctx->queue;
+        id<MTLCommandBuffer>      cmd_buf = [queue commandBuffer];
+        id<MTLBlitCommandEncoder> encoder = [cmd_buf blitCommandEncoder];
+        [cmd_buf enqueue];
+
+        size_t buf_dst_offset = 0;
+        id<MTLBuffer> buf_dst = ggml_metal_get_buffer(tensor, &buf_dst_offset);
+
+        buf_dst_offset += offset;
+
+        [encoder fillBuffer:buf_dst
+                      range:NSMakeRange(buf_dst_offset, buf_dst_offset + size)
+                      value:value];
+
+        [encoder endEncoding];
+
+        [cmd_buf commit];
+        [cmd_buf waitUntilScheduled];
```

Comment:
It is not obvious to me why `waitUntilScheduled` is necessary here, from what I can tell this is something that only needs to be done on iOS when the app is about to be moved to the background.

Reply:
Was worried that exiting the `autoreleasepool` section before fully scheduled might be a problem, but my guess it that the it's destructor likely has this call in it.